core/src/main/java/org/radargun/sysmonitor/ThreadDumpMonitor.java

```diff
@@ -0,0 +1,84 @@
+package org.radargun.sysmonitor;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.radargun.reporting.Timeline;
+import org.radargun.traits.JmxConnectionProvider;
+
+import static java.lang.management.ManagementFactory.THREAD_MXBEAN_NAME;
+import static java.lang.management.ManagementFactory.newPlatformMXBeanProxy;
+
+public class ThreadDumpMonitor extends JmxMonitor {
+
+   private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mm:ss");
+
+   private ThreadMXBean threadMXBean;
+   private final boolean lockedMonitors;
+   private final boolean lockedSynchronizers;
+   private final int workerIndex;
+
+
+   public ThreadDumpMonitor(JmxConnectionProvider jmxConnectionProvider, Timeline timeline,
+                            boolean dumpThreadLockedMonitors, boolean dumpThreadLockedSynchronizers, int workerIndex) {
+      super(jmxConnectionProvider, timeline);
+      this.lockedMonitors = dumpThreadLockedMonitors;
+      this.lockedSynchronizers = dumpThreadLockedSynchronizers;
+      this.workerIndex = workerIndex;
+   }
+
+   public synchronized void runMonitor() {
+      try {
+         if (connection == null) {
+            log.warn("MBean connection is not open, cannot read Thread Dump");
+            return;
+         }
+         threadDump();
+      } catch (Exception e) {
+         log.error("Exception!", e);
+      }
+   }
+
+   @Override
+   public synchronized void start() {
+      super.start();
+      try {
+         this.threadMXBean = newPlatformMXBeanProxy(connection, THREAD_MXBEAN_NAME, ThreadMXBean.class);
+      } catch (IOException e) {
+         throw new IllegalStateException("Cannot create threadMXBean", e);
+      }
+   }
+
+   @Override
+   public synchronized void stop() {
+      super.stop();
+   }
+
+   public void threadDump() {
+      log.info("Generating thread dump for: " + this.workerIndex);
+      String fileName = "thread-dump-worker-" + this.workerIndex + "-" + DATE_FORMATTER.format(LocalDateTime.now()) + ".dump";
+      BufferedWriter writer = null;
+      try {
+         writer = new BufferedWriter(new FileWriter(fileName, true));
```

Comment:
You should use try-with-resources it's more safe and you can delete all the finally block